[ICU] build failure

### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
vcpkg install icu
```


### Failure logs

C:\vcpkg>vcpkg install icu
warning: The vcpkg C:\vcpkg\vcpkg.exe is using detected vcpkg root C:\vcpkg and ignoring mismatched VCPKG_ROOT environment value C:\software\tools\vcpkg. To suppress this message, unset the environment variable or use the --vcpkg-root command line switch.
warning: In the September 2023 release, the default triplet for vcpkg libraries changed from x86-windows to the detected host triplet (x64-windows). For the old behavior, add --triplet x86-windows . To suppress this message, add --triplet x64-windows .
Computing installation plan...
The following packages will be built and installed:
    icu[core,tools]:x64-windows -> 73.1
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\heimann\AppData\Local\vcpkg\archives in 287 us. Use --debug to see more details.
Installing 1/1 icu:x64-windows...
Building icu[core,tools]:x64-windows...
-- Using cached icu4c-73_1-src.tgz.
-- Cleaning sources at C:/vcpkg/buildtrees/icu/src/c-73_1-src-3575e21c7c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/icu4c-73_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Using source at C:/vcpkg/buildtrees/icu/src/c-73_1-src-3575e21c7c.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Using cached msys2-autoconf-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-automake-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-binutils-2.40-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys2-make-4.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-pkgconf-1.9.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-which-2.21-4-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf-archive-2022.09.03-1-any.pkg.tar.zst.
-- Using cached msys2-bash-5.2.015-1-x86_64.pkg.tar.zst.
-- Using cached msys2-coreutils-8.32-5-x86_64.pkg.tar.zst.
-- Using cached msys2-file-5.44-5-x86_64.pkg.tar.zst.
-- Using cached msys2-gawk-5.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-grep-1~3.0-6-x86_64.pkg.tar.zst.
-- Using cached msys2-gzip-1.12-2-x86_64.pkg.tar.zst.
-- Using cached msys2-diffutils-3.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-sed-4.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf2.71-2.71-1-any.pkg.tar.zst.
-- Using cached msys2-automake1.16-1.16.5-1-any.pkg.tar.zst.
-- Using cached msys2-libiconv-1.17-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libintl-0.21-2-x86_64.pkg.tar.zst.
-- Using cached msys2-zlib-1.2.13-1-x86_64.pkg.tar.zst.
-- Using cached msys2-tar-1.34-3-x86_64.pkg.tar.zst.
-- Using cached msys2-findutils-4.9.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-gmp-6.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-gcc-libs-11.3.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-libbz2-1.0.8-4-x86_64.pkg.tar.zst.
-- Using cached msys2-liblzma-5.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libzstd-1.5.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libreadline-8.2.001-3-x86_64.pkg.tar.zst.
-- Using cached msys2-mpfr-4.2.0-2-x86_64.pkg.tar.zst.
-- Using cached msys2-libpcre-8.45-3-x86_64.pkg.tar.zst.
-- Using cached msys2-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys2-perl-5.36.0-1-x86_64.pkg.tar.zst.
-- Using cached msys2-ncurses-6.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libcrypt-2.1-4-x86_64.pkg.tar.zst.
-- Using msys root at C:/vcpkg/downloads/tools/msys2/87962f0d86055c95
CMake Warning at scripts/cmake/vcpkg_configure_make.cmake:269 (message):
  Unable to find system dir in the PATH variable! Appending required msys
  paths!
Call Stack (most recent call first):
  ports/icu/portfile.cmake:54 (vcpkg_configure_make)
  scripts/ports.cmake:168 (include)


-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using msys root at C:/vcpkg/downloads/tools/msys2/6f3fa1a12ef85a6f
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
-- Installing x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/vcpkg/downloads/tools/msys2/87962f0d86055c95/usr/bin/make.exe -j 17 --trace -f Makefile install DESTDIR=/C/vcpkg/packages/icu_x64-windows
    Working Directory: C:/vcpkg/buildtrees/icu/x64-windows-dbg/
    See logs for more information:
      C:\vcpkg\buildtrees\icu\install-x64-windows-dbg-out.log
      C:\vcpkg\buildtrees\icu\install-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:168 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/icu/portfile.cmake:144 (vcpkg_install_make)
  scripts/ports.cmake:168 (include)


error: building icu:x64-windows failed with: BUILD_FAILED
Elapsed time to handle icu:x64-windows: 6.5 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[icu]+Build+error&body=Copy+issue+body+from+C%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

### Additional context

This error has already been fixed in the https://github.com/microsoft/vcpkg/issues/32711

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -235,22 +235,15 @@ function(vcpkg_configure_make)
         cmake_path(CONVERT "$ENV{PATH}" TO_CMAKE_PATH_LIST path_list NORMALIZE)
         cmake_path(CONVERT "$ENV{SystemRoot}" TO_CMAKE_PATH_LIST system_root NORMALIZE)
         cmake_path(CONVERT "$ENV{LOCALAPPDATA}" TO_CMAKE_PATH_LIST local_app_data NORMALIZE)
+        string(TOUPPER "${local_app_data}" local_app_data_upper)
         file(REAL_PATH "${system_root}" system_root)
         string(TOUPPER "${system_root}" system_root_upper)
 
         message(DEBUG "path_list:${path_list}") # Just to have --trace-expand output
 
         vcpkg_list(SET find_system_dirs 
-            "${system_root}/system32"
-            "${system_root}/System32"
-            "${system_root}/system32/"
-            "${system_root}/System32/"
-            "${system_root_upper}/system32"
-            "${system_root_upper}/System32"
-            "${system_root_upper}/system32/"
-            "${system_root_upper}/System32/"
-            "${local_app_data}/Microsoft/WindowsApps"
-            "${local_app_data}/Microsoft/WindowsApps/"
+            "${system_root_upper}/SYSTEM32"
+            "${local_app_data_upper}/MICROSOFT/WINDOWSAPPS"
         )
 
         string(TOUPPER "${find_system_dirs}" find_system_dirs_upper)
@@ -258,7 +251,8 @@ function(vcpkg_configure_make)
         set(index 0)
         set(appending TRUE)
         foreach(item IN LISTS path_list)
-            if(item IN_LIST find_system_dirs OR item IN_LIST find_system_dirs_upper)
+            string(TOUPPER "${item}" item_upper)
+            if(item_upper IN_LIST find_system_dirs_upper OR "${item_upper}/" IN_LIST find_system_dirs_upper)
                 set(appending FALSE)
                 break()
             endif()

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -235,15 +235,15 @@ function(vcpkg_configure_make)
         cmake_path(CONVERT "$ENV{PATH}" TO_CMAKE_PATH_LIST path_list NORMALIZE)
         cmake_path(CONVERT "$ENV{SystemRoot}" TO_CMAKE_PATH_LIST system_root NORMALIZE)
         cmake_path(CONVERT "$ENV{LOCALAPPDATA}" TO_CMAKE_PATH_LIST local_app_data NORMALIZE)
-        string(TOUPPER "${local_app_data}" local_app_data_upper)
         file(REAL_PATH "${system_root}" system_root)
-        string(TOUPPER "${system_root}" system_root_upper)
 
         message(DEBUG "path_list:${path_list}") # Just to have --trace-expand output
 
         vcpkg_list(SET find_system_dirs 
-            "${system_root_upper}/SYSTEM32"
-            "${local_app_data_upper}/MICROSOFT/WINDOWSAPPS"
+            "${system_root}/System32"
+            "${system_root}/System32/"
+            "${local_app_data}/Microsoft/WindowsApps"
+            "${local_app_data}/Microsoft/WindowsApps/"
         )
 
         string(TOUPPER "${find_system_dirs}" find_system_dirs_upper)
@@ -252,7 +252,7 @@ function(vcpkg_configure_make)
         set(appending TRUE)
         foreach(item IN LISTS path_list)
             string(TOUPPER "${item}" item_upper)
-            if(item_upper IN_LIST find_system_dirs_upper OR "${item_upper}/" IN_LIST find_system_dirs_upper)
+            if(item_upper IN_LIST find_system_dirs_upper)
                 set(appending FALSE)
                 break()
             endif()

